### ISSUE TITLE
Methods from a parent class are not shown when the base class is a class attribute of another class

### ISSUE THREAD

**Issue by ThiefMaster** [microsoft/pyright#2044]
**Describe the bug**
Methods from a parent class are not shown when the base class is a class attribute of another class.

**To Reproduce**
Just paste the code below in a file

**Expected behavior**
`MyModel` having autocomplete for `foo`, and `MyModel.foo` having the proper type from the `foo` method:

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
class MyModelBase:
    @classmethod
    def foo(cls):
        pass


class MyDBClass:
    Model: MyModelBase

    def __init__(self):
        # This `__init__` method is part of a library in my real code, hence the
        # type hint above. but even when i'm super explicit here it doesn't work
        self.Model = MyModelBase


db = MyDBClass()


class MyModel(db.Model):
    pass


MyModel.foo
```

![image](https://user-images.githubusercontent.com/179599/124181653-18ce2c00-dab6-11eb-81c1-4902b14f2d12.png)

**VS Code extension or command-line**
pyright 1.1.154 via Pylance in the latest vscode insiders

**Additional context**
The real code is using Flask-SQLAlchemy with a custom model_class and a custom subclass of `SQLAlchemy` that adds some convenience methods to all models.

**Comment by erictraut** [comment]
You have a bug in the type annotation for the class variable `Model`. You've annotated it to accept an _instance_ of `MyModelBase`, but then you're assigning the `MyModelBase` class to it. The annotation should be `Model: Type[MyModelBase]` if your intent is that `Model` can contain an instantiable class (as opposed to a class instance).

Also, you are using a dynamic expression in the class definition for MyModel. Static type checkers like pyright or mypy need to understand class hierarchies, so they do not support dynamic class declarations like this.

**Comment by ThiefMaster** [comment]
Indeed, `Model: Type[MyModelBase]` fixed it both in my example and in my real application - thanks!
